code/model/modelanimation.cpp

```diff
@@ -1193,6 +1246,35 @@ namespace animation {
 		s_animationsById.emplace(helper.m_animationName, ParsedModelAnimation { animation, type, name, subtype });
 	}
 
+	void ModelAnimationParseHelper::parseSingleMoveable() {
+		volatile ModelAnimationMoveableOrientation o = ModelAnimationMoveableOrientation(nullptr, angles{ 0,0,0 });
+		volatile ModelAnimationMoveableRotation r = ModelAnimationMoveableRotation(nullptr, angles{ 0,0,0 }, angles{0,0,0}, optional<angles>());
+		
+		required_string("$Name:");
+		char animID[NAME_LENGTH];
+		stuff_string(animID, F_NAME, NAME_LENGTH);
+
+		SCP_string name = animID;
+
+		if (s_moveablesById.count(name))
+			error_display(1, "Animation with name %s already exists!", animID);
+
+		required_string("$Type:");
+
+		char segment_type[NAME_LENGTH];
+		stuff_string(segment_type, F_NAME, NAME_LENGTH);
+
+		auto segment_parser = s_moveableParsers.find(segment_type);
+		if (segment_parser != s_moveableParsers.end()){
+			s_moveablesById.emplace(name, segment_parser->second());
+		}
+		else {
+			error_display(1, "Unknown moveable type %s in moveable with ID %s!", segment_type, animID);
```

Comment:
That's two different cases.
Moveable with name not found means that someone tried to attach a moveable to a ship, but the animation was not found. Currently, that is handled by just ignoring it.
Unknown moveable type means that someone tried to define a moveable, but the definition itself is ill-formed.

Technically I could make that a warning as well, and then just not create the animation, but that would cause a follow-up warning of "Moveable not found" (as its creation was just aborted), so the cause of the problem might not be immediately obvious. _If_ we want both problems to be of the same type, then I'd suggest to also hard error if someone tried to add a moveable to a ship which does not exist.

Reply:
Nah, I didn't fully grasp what was going on.  Your explanation helps and it makes sense to keep the different warning levels as they are.

I would suggest fleshing out the messages a bit more though.  For this one you could say "Could not create moveable; %s is not a valid type" or something like that